eternalcore-core/src/main/java/com/eternalcode/core/feature/vanish/controller/ItemController.java

```diff
@@ -0,0 +1,62 @@
+package com.eternalcode.core.feature.vanish.controller;
+
+import com.eternalcode.core.configuration.implementation.PluginConfiguration;
+import com.eternalcode.core.feature.vanish.VanishConfig;
+import com.eternalcode.core.feature.vanish.VanishService;
+import com.eternalcode.core.injector.annotations.Inject;
+import com.eternalcode.core.injector.annotations.component.Controller;
+import com.eternalcode.core.notice.NoticeService;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.entity.EntityPickupItemEvent;
+import org.bukkit.event.player.PlayerDropItemEvent;
+
+@Controller
+class ItemController implements Listener {
+
+    private final NoticeService noticeService;
+    private final VanishService vanishService;
+    private final VanishConfig config;
+
+    @Inject
+    ItemController(NoticeService noticeService, VanishService vanishService, PluginConfiguration config) {
+        this.noticeService = noticeService;
+        this.vanishService = vanishService;
+        this.config = config.vanish;
+    }
+    
+    @EventHandler
+    void onPickUp(EntityPickupItemEvent event) {
+        if (!this.config.blockItemPickup) {
+            return;
+        }
+        if (!(event.getEntity() instanceof Player player)) {
+            return;
+        }
+
+        if (!this.vanishService.isVanished(player)) {
+            return;
+        }
+        
+        event.setCancelled(true);
+
+        this.noticeService.player(player.getUniqueId(), message -> message.vanish().cantPickupItemsWhileVanished());
```

Comment:
unnecessary imo, some people will trigger this accidentally, for example if you walk into a pile of items while vanished you will get spammed chat

Reply:
User can easy configure boolean or message - etc just change notice to null im cfg